### PR TITLE
chore: Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

---
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/argo-rollouts/commit/e1d381bf474944d9786c8383cad5a7e15d3e3b37#commitcomment-42675752

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/argo-rollouts/commit/d414d0caf920ad58e454be17177608ff8a9d9d68

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: analysis/analysis.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ func (c *Controller) reconcileAnalysisRun(origRun *v1alpha1.AnalysisRun) *v1alph
 		if enqueueSeconds < 0 {
 			enqueueSeconds = 0
 		}
-		log.Infof("enqueuing analysis after %v", enqueueSeconds)
+		log.Infof("enqueueing analysis after %v", enqueueSeconds)
```
Comment (jsoref) [inline analysis/analysis.go:103]: I'm not attached to one spelling or the other, but this repo was using both.

File: docs/features/analysis.md [hunk 2]
```diff
@@ -708,13 +708,13 @@ This example demonstrates:
         spec:
           containers:
           - name: guestbook
-            image: guesbook:v1
+            image: guestbook:v1
```
Comment (jsoref) [inline docs/features/analysis.md:711]: Not sure about this change. I hope it's just a typo...

File: docs/getting-started/istio/index.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ spec:
   hosts:
   - rollouts-demo.local
   http:
-  - name: primary  # Should match spec.strategy.canary.stableSertrafficRouting.istio.virtualService.routes
+  - name: primary  # Should match spec.strategy.canary.trafficRouting.istio.virtualService.routes
```
Comment (jsoref) [inline docs/getting-started/istio/index.md:58]: This is by far the oddest change in the PR

File: metricproviders/kayenta/kayenta.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ const (
                             }`
 
 	resumeDelay          time.Duration = 15 * time.Second
-	httpConnectionTimout time.Duration = 15 * time.Second
+	httpConnectionTimeout time.Duration = 15 * time.Second
```
Comment (jsoref) [inline metricproviders/kayenta/kayenta.go:43]: I'm sure there's a go linter that hates me

File: rollout/replicaset_test.go [hunk 5]
```diff
@@ -34,10 +34,10 @@ func int32Ptr(i int32) *int32 { return &i }
 func TestGetReplicaSetsForRollouts(t *testing.T) {
 	newTimestamp := metav1.Date(2016, 5, 20, 2, 0, 0, 0, time.UTC)
 	selector := map[string]string{
-		"app": "ngnix",
+		"app": "nginx",
 	}
 	diffSelector := map[string]string{
-		"app": "ngnix2",
+		"app": "nginx2",
```
Comment (jsoref) [inline rollout/replicaset_test.go:40]: I assume this doesn't matter

File: rollout/restart.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ const (
 	// restartPodCheckTime prevents the Rollout from not making any progress with restarting Pods. When pods can be restarted
 	// faster than the old pods can be scaled down, the parent's ReplicaSet's availableReplicas does not change. A rollout
 	// relies on changes to the availableReplicas of the ReplicaSet to detect when the controller should requeue and continue
-	// deleting pods. In this situation, the rollout does not renqueue and wont make any more progress restarting pods until
+	// deleting pods. In this situation, the rollout does not requeue and won't make any more progress restarting pods until
```
Comment (jsoref) [inline rollout/restart.go:20]: The first change here is a bit odd. I considered `re-enqueue`, but `requeue` is used elsewhere.

File: utils/kubeclientmetrics/metric_test.go [hunk 7]
```diff
@@ -106,7 +106,7 @@ func TestResolveK8sRequestVerb(t *testing.T) {
 		assert.Equal(t, List, verb)
 	})
 	t.Run("VirtualService GET", func(t *testing.T) {
-		r := request("https://127.0.0.1/apis/networking.istio.io/v1alpha3/namespaces/default/virtualservices/virutal-service")
+		r := request("https://127.0.0.1/apis/networking.istio.io/v1alpha3/namespaces/default/virtualservices/virtual-service")
```
Comment (jsoref) [inline utils/kubeclientmetrics/metric_test.go:109]: Uncertain about this change. Hoping it doesn't matter

File: utils/replicaset/replicaset_test.go [hunk 8]
```diff
@@ -621,7 +621,7 @@ func TestMaxUnavailable(t *testing.T) {
 }
 
 func TestCheckPodSpecChange(t *testing.T) {
-	ro := generateRollout("ngnix")
+	ro := generateRollout("nginx")
```
Comment (jsoref) [inline utils/replicaset/replicaset_test.go:624]: puzzled by this change

File: docs/getting-started/istio/index.md (hunk 3 above)
Comment (jessesuen) [reply docs/getting-started/istio/index.md:58]: 🤣 looks like I wanted to say stableService, then had a mild stroke before remembering it should be trafficRouting.

**Review by jessesuen — APPROVED** [review]
Wow. Thanks for this!

**Pull request merged**